src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
+            // result in an exception.
+
+            /// <summary>
+            /// The task returned from the last SendAsync operation to not complete synchronously.
+            /// If this is not null and not completed when a subsequent SendAsync is issued, an exception occurs.
+            /// </summary>
+            private Task _lastSendAsync;
+            /// <summary>
+            /// The task returned from the last ReceiveAsync operation to not complete synchronously.
+            /// If this is not null and not completed when a subsequent ReceiveAsync is issued, an exception occurs.
+            /// </summary>
+            private Task<WebSocketReceiveResult> _lastReceiveAsync;
+            /// <summary>
+            /// Semaphore used to ensure that calls to SendFrameAsync don't run concurrently.  While <see cref="_lastSendAsync"/>
+            /// is used to fail if a caller tries to issue another SendAsync while a previous one is running, internally
+            /// we use SendFrameAsync as an implementation detail, and it should not cause user requests to SendAsync to fail,
+            /// nor should such internal usage be allowed to run concurrently with other internal usage or with SendAsync.
+            /// </summary>
+            private readonly SemaphoreSlim _sendFrameAsyncLock = new SemaphoreSlim(1, 1);
+
+            public ManagedClientWebSocket()
             {
-                throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
+                // Initialize the keep alive timer.  When active, it'll send periodic ping frames to the server.
+                _keepAliveTimer = new Timer(
+                    s => ((ManagedClientWebSocket)s).SendKeepAliveFrameAsync(), this, 
+                    Timeout.Infinite, Timeout.Infinite);
+
+                // Set up the abort source so that if it's triggered, we transition the instance appropriately
+                _abortSource.Token.Register(s =>
+                {
+                    var thisRef = (ManagedClientWebSocket)s;
+
+                    WebSocketState state = thisRef._state;
+                    if (state != WebSocketState.Closed && state != WebSocketState.Aborted)
+                    {
+                        thisRef._state = state != WebSocketState.None && state != WebSocketState.Connecting ?
+                            WebSocketState.Aborted :
+                            WebSocketState.Closed;
+                    }
+                }, this);
             }
-        }
 
-        public WebSocketCloseStatus? CloseStatus
-        {
-            get
+            public override void Dispose()
             {
-                throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
+                if (!_disposed)
+                {
+                    _disposed = true;
+                    _keepAliveTimer.Dispose();
+                    _stream?.Dispose();
+                    _client?.Dispose();
```

Comment:
I considered that, but opted not to as the main ClientWebSocket class does the inverse: Abort calls Dispose, but Dispose doesn't call Abort:
https://github.com/stephentoub/corefx/blob/b65762052589009332599af9d8a4585436f6d6fc/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocket.cs#L217-L242